### ISSUE TITLE
GetThreadHistory Parameters

### ISSUE THREAD

**Issue by AllanWang** [Schmavery/facebook-chat-api#262]
There's a change I might not have understood the function, but regardless of what I set for "start" and "end", it will always read the messages starting from the most recent ones.

[Here ](https://github.com/AllanWang/AllanBot-Public/blob/5bf3766389b3c7cd8c2992856b29e857ce5df04c/src/quote.js)is my code when I was testing it.

From my understanding, if I set start to 5 and end to 10, it would read the 5th to 10th last messages.
For all my tests, the timestamp was set to Date.now() and I used various numbers for start and end, but they all checked the same messages.

**Comment by bsansouci** [comment]
Try setting timestamp to `null` instead of `Date.now()`. If `timestamp` is set, `start` is effectively ignored.

**Comment by AllanWang** [comment]
I tried that as well and it doesn't seem to change anything.

**Comment by bsansouci** [comment]
Aaah crap. Offset is broken on my side too @Schmavery :(
I'll look more into it tomorrow

**Comment by alexisohayon** [comment]
Same issue here :) How can I fix this on my side?

**Comment by Schmavery** [comment]
This issue has been seen a number of times in different forms.  Add I've said in previous related issues, either Facebook has changed some signature or the docs are wrong. Someone should check again what request the browser is sending to retrieve old messages.

**Comment by GAMELASTER** [comment]
```
messages[user_ids][1000029***48239][offset]:32
messages[user_ids][1000029***48239][timestamp]:1461849806535
messages[user_ids][1000029***48239][limit]:32
```
This works for me nice, where the timestamp is the time of the "oldest message"

**Comment by bsansouci** [comment]
@GAMELASTER can you find out how we could query the messages without a timestamp? I'd be much nicer to query the last 10 messages, and then the 10 messages after that etc... It's pretty likely that they don't allow that anymore because on their front-end they can just take the timestamp of the last message they received when querying for 0-10 and use it for 10-20 right?

**Comment by Schmavery** [comment]
Should be closed by #453?
